File: C++/3Sum-Closest.cpp
class Solution {
public:
    int threeSumClosest(vector<int>& nums, int target) {
        sort(nums.begin(), nums.end());
        int ClosestVal = INT_MAX, sum = 0;
        for (int i = 0; i + 2 < nums.size(); i ++) {
            int left = i + 1, right = nums.size() - 1;
            while (left < right) {
                int temp = nums[left] + nums[right] + nums[i];
                if (ClosestVal > abs(temp - target)) {
                    ClosestVal = abs(temp - target);
                    sum = temp;
                }
                if (temp == target) return target;
                else if (temp < target) {
                    left ++;
                } else {
                    right --;
                }
            }
        }
        return sum;
    }
};

File: Python/3Sum-Closest.py
class Solution:
    # @return an integer
    def threeSumClosest(self, num, target):
        num.sort()
        res = num[0] + num[1] + num[2]
        for i in range(len(num)-2):
            left = i + 1
            right = len(num) - 1
            while left < right:
                s = num[i] + num[left] + num[right]
                if abs(s-target) < abs(res-target):
                    res = s
                if s == target:
                    return s
                elif s < target:
                    left += 1
                else:
                    right -= 1
        return res